### PR TITLE
✨ Source auth0: add new stream `clients`

### PR DESCRIPTION
**Pull request by vasilisgav**
## What
*Add support for clients in auth0 source connector* issue #25951

## How
*Connector in place had all the scaffolding ready, so adding a new stream was simple enough*

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Actions

<details><summary><strong>New Stream for source Connector auth0</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.2.0`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-auth0/source_auth0/source.py [hunk 1]
```diff
@@ -113,12 +113,18 @@ def request_params(
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
         entities = response.json()
+        print("VG 118")
+        print(entities)
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-auth0/source_auth0/source.py:117]: Remove this prints.

File: airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py [hunk 2]
```diff
@@ -239,3 +239,23 @@ def test_users_source_parse_response(self, requests_mock, patch_base_class, user
...
+
+class TestStreamClients:
+    def test_stream_clients(self, patch_base_class, clients_instance, url_base, api_url, requests_mock):
+        stream = Clients(url_base=url_base)
+        requests_mock.get(
+            f"{api_url}/clients",
+            json={"total": 1, "start": 0, "limit": 50, "clients": [clients_instance]},
+        )
+        inputs = {"sync_mode": SyncMode.full_refresh}
+        assert list(stream.read_records(**inputs)) == [clients_instance]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py:252]: Thanks this is really helpful!

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Small details but nice!

File: airbyte-integrations/connectors/source-auth0/source_auth0/source.py (hunk 1 above)
Comment (vasilisgav) [reply airbyte-integrations/connectors/source-auth0/source_auth0/source.py]: thanks!

**Comment by vasilisgav** [comment]
running unit tests
```
python -m pytest unit_tests
Test session starts (platform: darwin, Python 3.9.16, pytest 6.2.5, pytest-sugar 0.9.7)
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/vgavriil/workspace/forks/airbyteAuth0/airbyte-integrations/connectors/source-auth0/.hypothesis/examples')
rootdir: /Users/vgavriil/workspace/forks/airbyteAuth0, configfile: pytest.ini
plugins: hypothesis-6.54.6, requests-mock-1.9.3, mock-3.6.1, sugar-0.9.7, timeout-1.4.2, cov-3.0.0
collected 30 items

 airbyte-integrations/connectors/source-auth0/unit_tests/test_source.py::TestAuthentication.test_init_token_authentication_init ✓                                3% ▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_source.py::TestAuthentication.test_init_oauth2_authentication_init ✓                               7% ▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_source.py::TestAuthentication.test_init_oauth2_authentication_wrong_credentials_record ✓          10% █
 airbyte-integrations/connectors/source-auth0/unit_tests/test_source.py::TestAuthentication.test_init_oauth2_authentication_wrong_oauth_config_bad_auth_type ✓  13% █▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_source.py::TestAuthentication.test_check_connection_ok ✓                                          17% █▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_source.py::TestAuthentication.test_check_connection_error_status_code ✓                           20% ██
 airbyte-integrations/connectors/source-auth0/unit_tests/test_source.py::TestAuthentication.test_check_connection_error_with_exception ✓                        23% ██▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_source.py::TestAuthentication.test_check_streams ✓                                                27% ██▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_auth0_stream_request_params ✓                                    30% ███
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_auth0_stream_parse_response ✓                                    33% ███▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_auth0_stream_backoff_time ✓                                      37% ███▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_auth0_stream_incremental_request_params ✓                        40% ████
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_incremental_auth0_stream_parse_response ✓                        43% ████▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_incremental_auth0_stream_backoff_time ✓                          47% ████▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_auth0_stream_incremental_backoff_time_empty ✓                    50% █████
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_auth0_stream_incremental_back_off_now ✓                          53% █████▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_auth0_stream_incremental_get_updated_state ✓                     57% █████▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestAuth0Stream.test_auth0_stream_http_method ✓                                       60% ██████
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestNextPageToken.test_next_page_token ✓                                              63% ██████▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestNextPageToken.test_next_page_token_invalid_cursor ✓                               67% ██████▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestNextPageToken.test_next_page_token_missing_cursor ✓                               70% ███████
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestNextPageToken.test_next_page_token_one_page_only ✓                                73% ███████▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestNextPageToken.test_next_page_token_last_page_incomplete ✓                         77% ███████▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestNextPageToken.test_next_page_token_last_page_complete ✓                           80% ████████
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestStreamUsers.test_stream_users ✓                                                   83% ████████▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestStreamUsers.test_users_request_params_out_of_next_page_token ✓                    87% ████████▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestStreamUsers.test_users_source_request_params_have_next_cursor ✓                   90% █████████
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestStreamUsers.test_users_source_parse_response ✓                                    93% █████████▍
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestStreamClients.test_stream_clients ✓                                               97% █████████▋
 airbyte-integrations/connectors/source-auth0/unit_tests/test_streams.py::TestStreamClients.test_clients_source_parse_response ✓                               100% ██████████

```

**Comment by marcosmarxm** [comment]
This was published in https://github.com/airbytehq/airbyte/pull/26445

**Pull request closed without merging**